### PR TITLE
Use ubuntu-latest image in GitHub Actions

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

`PyVista` and `vtk` are `Python` tools and are basically unaffected by differences in Ubuntu versions.
This change frees us from future Ubuntu version changes by using ubuntu-latest in GitHub Actions.
_Edit: Changes that will occur by setting the ubuntu version of the current GitHub Actions to the latest is listed below._

### Details

- [x] Do NOT Update test image file for the ubuntu-latest environment.

As shown below, ubuntu-22.04 (ubuntu-latest) is causing some parts of the wireframe to turn black.
| ubuntu 20.04 | ubuntu-latest |
|---|---|
| plot_list | plot_list (1) |

### CODE REVIEW

**Comment by banesullivan** [comment]
The rendering differences look like issues with the version of OSMesa. IMO, the new renderings are unexpected and we should not alter our ground-truth images to match. We may need to get to the root of what is causing these rendering differences and report upstream to VTK

**Comment by tkoyama010** [comment]
You are absolutely right about not updating the images. Thank you for setting out the policy to determine the cause.

**Comment by tkoyama010** [comment]
.pre-commit.ci autofix

**Comment by banesullivan** [comment]
@tkoyama010, please refer to #3740 for why `ubuntu-20.04` is used which is due to differences in the versions of Mesa shipped with each version. cc @akaszynski

File: tests/plotting/test_plotting.py [hunk 1]
```diff
@@ -264,6 +264,10 @@ def test_plot_update(sphere):
     pl.close()
 
 
+# https://github.com/pyvista/pyvista/pull/4950
+@pytest.mark.skipif(
+    platform.system() == 'Linux', reason='Does not display correctly within latest ubuntu'
+)
```
Comment (banesullivan) [inline tests/plotting/test_plotting.py:270]: We should avoid these sorts of skips and instead skip via the `verify_image_cache` fixture

**Comment by banesullivan** [comment]
I'm all for using the `ubuntu-latest` image but I want to validate that this isn't significantly slowing down our CI and that we aren't disabling too many tests

**Comment by tkoyama010** [comment]
@banesullivan 100%. Let's consider this a little more carefully.

**Comment by banesullivan** [comment]
Looks like `test_plot_cubemap_alone`. I vaguely recall tracking a failure with cubemaps in VTK with recent versions of Mesa... Perhaps: 

- https://gitlab.kitware.com/vtk/vtk/-/issues/18584
- https://github.com/pyvista/pyvista/pull/4688

**Comment by banesullivan** [comment]
@tkoyama010, would you be okay with closing this PR and tracking this in an issue? I think we should stick with our current CI configurations for the foreseeable future

**Comment by tkoyama010** [comment]
@banesullivan I agree with that. Closing this.

**Pull request closed without merging**